include/core/evaporation_model.h

```diff
@@ -0,0 +1,707 @@
+           std::exp(-L_vap_x_M_x_R_inv *
+                    (temperature_inv - boiling_temperature_inv));
+  }
+
+  /**
+   * @brief vector_mass_flux Calculates the values of the staturation pressure
+   * for multiple points
+   * @param field_vectors Value of the various fields on which the flux may depend.
+   * @param mass_flux_vector Vectors of the mass flux values
+   */
+  void
+  vector_saturation_pressure(
+    const std::map<field, std::vector<double>> &field_vectors,
+    std::vector<double>                        &saturation_pressure_vector)
+  {
+    const std::vector<double> &temperature =
+      field_vectors.at(field::temperature);
+
+    for (unsigned int i = 0; i < saturation_pressure_vector.size(); ++i)
+      {
+        const double temperature_inv = 1.0 / (temperature[i] + 1e-16);
+
+        saturation_pressure_vector[i] =
+          ambient_pressure *
+          std::exp(-L_vap_x_M_x_R_inv *
+                   (temperature_inv - boiling_temperature_inv));
+      }
+  }
+
+  /**
+   * @brief mass_flux Calculates the value of the saturated vapor pressure.
+   * @param fields_value Value of the various field on which the saturated
+   * vapor pressure may depend.
+   * @return value of the pressure calculated with the fields_value
+   */
+  double
+  mass_flux(const std::map<field, double> &fields_value) override
+  {
+    const double saturation_pressure_value = saturation_pressure(fields_value);
+    const double temperature_inv =
+      1.0 / (fields_value.at(field::temperature) + 1e-16);
+
+    const double mass_flux_value =
+      evaporation_coefficient * saturation_pressure_value *
+      std::sqrt(M_x_2PI_inv_x_R_inv * temperature_inv);
+
+    return mass_flux_value;
+  }
+
+  /**
+   * @brief vector_mass_flux Calculates the values of the evaporation mass flux
+   * for multiple points
+   * @param field_vectors Value of the various fields on which the flux may depend.
+   * @param mass_flux_vector Vectors of the mass flux values
+   */
+  void
+  vector_mass_flux(const std::map<field, std::vector<double>> &field_vectors,
+                   std::vector<double> &mass_flux_vector) override
+  {
+    const std::vector<double> &temperature =
+      field_vectors.at(field::temperature);
+
+    const unsigned int n_pts = mass_flux_vector.size();
+
+    std::vector<double> saturation_pressure_vector(n_pts);
+    vector_saturation_pressure(field_vectors, saturation_pressure_vector);
+
+    for (unsigned int i = 0; i < mass_flux_vector.size(); ++i)
+      {
+        const double temperature_inv = 1.0 / (temperature[i] + 1e-16);
+
+        mass_flux_vector[i] = evaporation_coefficient *
+                              saturation_pressure_vector[i] *
+                              std::sqrt(M_x_2PI_inv_x_R_inv * temperature_inv);
+      }
+  }
+
+  /**
+   * @brief heat_flux Calculates the value of the evaporation heat flux.
+   * @param fields_value Value of the various field on which the flux may depend.
```

Comment:
```suggestion
   * @param fields_value Value of the various fields on which the flux may depend.
```

Reply:
I think we say "various fields", so you could do a control find and replace all of these